backend/compact-connect/stacks/ingest_stack.py

```diff
@@ -61,6 +61,7 @@ def __init__(
             entry=os.path.join('lambdas', 'license-data'),
             index=os.path.join('handlers', 'ingest.py'),
             handler='process_license_message',
+            timeout=Duration.minutes(1),
```

Comment:
These lambdas are now picking up batches of up to 50 messages per invocation. Performance testing showed that they reliably finished in under 7 seconds. Under most failure scenarios, the lambda should gracefully report a set of messages that failed (if any failed), which allows the successful messages to be cleared off the queue and the failures to be retried later.

If the lambda starts timing out, things become much less efficient:
- The entire batch of 50 messages will be left on the queue and retried, regardless of whether they were processed successfully
- Frequent enough timeouts will trigger AWS to stop scaling the lambda to keep up with load on the SQS Queue, causing an increased backlog

Together, these two can create a self-reinforcing failure scenario that we want to avoid. By setting the timeout to 1 minute, I'm giving the lambda room to continue to consume messages, even if network issues or AWS service degradation slow it down by nearly a factor of 10. If the lambda is still not finishing after 10x the expected runtime, we may be better off just cutting it off and trying again. The messages will become visible to other consumers after 5 minutes.

Reply:
Thank you for the explanation!